src/modules/user/user.controller.ts

```diff
@@ -150,6 +153,108 @@ export class UserController {
     }
   }
 
+  @UseGuards(ThrottlerGuard)
+  @Throttle({
+    long: {
+      ttl: 3600,
+      limit: 5,
+    },
+  })
+  @Post('/')
+  @HttpCode(201)
+  @ApiOperation({
+    summary: 'Register Pre Created User',
+  })
+  @ApiOkResponse({ description: 'Creates Pre Created User' })
+  @ApiBadRequestResponse({ description: 'Missing required fields' })
+  @Public()
+  async registerPreCreatedUser(
+    @Body()
+      { invitationId, ...createUserDto }: RegisterPreCreatedUserDto,
+    @Req() req: Request,
+    @Res({ passthrough: true }) res: Response,
+  ) {
+    try {
+      const preCreatedUser = await this.userUseCases.findPreCreatedByEmail(
+        createUserDto.email,
+      );
+
+      if (!preCreatedUser) {
+        throw new NotFoundException('PRE_CREATED_USER_NOT_FOUND');
+      }
+
+      const userCreated = await this.userUseCases.createUser(createUserDto);
+
+      const keys = await this.keyServerUseCases.addKeysToUser(
+        userCreated.user.id,
+        createUserDto,
+      );
+
+      await this.userUseCases.replacePreCreatedUser(
+        userCreated.user.email,
+        userCreated.user.uuid,
+        keys.publicKey,
+      );
+
+      await this.sharingService.acceptInvite(userCreated.user, invitationId, {
+        encryptionKey: createUserDto.publicKey,
+        //TODO: What properties should be here?
+        encryptionAlgorithm: '',
```

Comment:
@sg-gs  @apsantiso please tell me what property should be here?

Reply:
The invite has the encryption key and algorithm in this case, dont send them. The user is not inviting himself to the sharing.

`    const newSharing = Sharing.build({
      ...invite,
      id: v4(),
      type: SharingType.Private,
      ownerId: owner.uuid,
      encryptionAlgorithm: invite.isARequest()
        ? acceptInviteDto.encryptionAlgorithm
        : invite.encryptionAlgorithm,
      encryptionKey: invite.isARequest()
        ? acceptInviteDto.encryptionKey
        : invite.encryptionKey,
    });`

Reply:
perfect, thanks!